### PR TITLE
Second report from same input file: data is generated

### PR DESCRIPTION
**Pull request by gaurav-dogra**
testing of the generated data and capability to download two reports pending

### CODE REVIEW

**Review by alfabravo2013 — APPROVED** [review]
MyController is not very good name. Give it some fancy name instead.

File: src/main/java/gmailgdogra/service/ShiftReportGeneratingService.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.List;
+
+public class ShiftReportGeneratingService {
+
+    public static XSSFWorkbook write(List<OutputRow> inSwipesCurrentShift) throws IOException {
+        FileInputStream templateFile = new FileInputStream(new File("src/main/resources/Shift Report Template.xlsx"));
```
Comment (alfabravo2013) [inline src/main/java/gmailgdogra/service/ShiftReportGeneratingService.java:17]: A good idea is to have such values as this filepath in a separate file, maybe even in a configuration file which you can easily edit without recompiling the application. It's not necessary, but think about it.

File: src/main/java/gmailgdogra/service/ShiftReportGeneratingService.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        XSSFSheet sheet = workbook.getSheetAt(0);
+        insertShiftDate(sheet);
+        templateFile.close();
+        return workbook;
+    }
+
+    private static void insertShiftDate(XSSFSheet sheet) {
+        LocalDateTime localDateTime = LocalDateTime.now();
+        System.out.println("localDateTime = " + localDateTime);
+        Row dateRow = sheet.getRow(3);
```
Comment (alfabravo2013) [inline src/main/java/gmailgdogra/service/ShiftReportGeneratingService.java:28]: Remember, named constants are always better than just numbers.

File: src/test/java/gmailgdogra/SwipeProcessorServiceTest.java [hunk 3]
```diff
@@ -165,21 +191,43 @@ private MockMultipartFile getFile(String pathname) throws IOException {
...
+    }
+
+    @Test
+    void testCurrentShiftSwipeInsForFile1() throws IOException {
+        MockMultipartFile testFile2 = getFile("src/main/resources/test file 1.xlsx");
+        List<SwipeRecord> allSwipes = readXlsxService.readAllRows(testFile2.getInputStream());
+        List<Shift> shifts = getShiftsForTestFile1();
+        swipeProcessorService.prepareData(allSwipes, shifts);
+        List<OutputRow> results = swipeProcessorService.getInSwipesCurrentShift();
+        assertTrue(results.containsAll(inSwipesCurrentShiftFromFile1), "Current Shift Swipe-in not as expected");
```
Comment (alfabravo2013) [inline src/test/java/gmailgdogra/SwipeProcessorServiceTest.java:220]: better to insert an empty line between the preparation and the assertion parts, to make it easier to identify visually

**Review by alfabravo2013 — APPROVED** [review]
Good job

File: src/main/java/gmailgdogra/service/ShiftReportGeneratingService.java (hunk 1 above)
Comment (gaurav-dogra) [reply src/main/java/gmailgdogra/service/ShiftReportGeneratingService.java]: Will have to find out how to best achieve it

**Pull request merged**